market-test/src/main/java/com/axonivy/market/MarketMetaSchema.java

```diff
@@ -84,6 +84,17 @@ public static enum ArtifactType {
     }
     public ArtifactType type;
     public boolean doc;
+
+    public static class ArchivedArtifact{
+      @NotNull @Size(min = 5)
+      public String lastVersion;
```

Comment:
this property has a confusing name for me. What does the last version do? How does it work. What values does it expect?
in json-schema you can provide such hints using the @JsonPropertyDescription annotation
com.fasterxml.jackson.annotation.JsonPropertyDescription

Reply:
Hi @ivy-rew 
Before the PR of https://github.com/axonivy-market/market.axonivy.com/pull/36
artifacts selector in Portal displays 2 versions of PortalApp ( PortalApp and PortalApp (10.0)) - and the same for portal guide because there are 2 group id used: from version 11.0.0++, they used group id of `com.axonivy.portal` while `ch.ivyteam.ivy.project.portal` is used for other
![image](https://github.com/axonivy-market/market/assets/127725498/4655e1b5-0eba-42b5-99f2-f5f37b983244)

It confuses users when they can not download PortalApp (10.0) for version 11.0.0++ and PortalApp if the version is before 11.0.0. So Portal team has asked us to remove these duplicates and 1 artifact for all versions.

We had a short discussion with Alex and it seemed that he agreed with the proposal of introducing new properties of archived artifacts in the schema to store old group id and artifact id.

The `lastVersion` property is designed to support "download" feature in the website: we need to detect which group id and artifact ID should be used to build the maven link:  
https://github.com/axonivy-market/market.axonivy.com/blob/776b98ddb080998808be9a6c3862ce6c42765351/src/app/domain/maven/MavenArtifact.php#L106
So, if the user chooses version 8.0 or 10.0.17, the maven URL with group id `ch.ivyteam.ivy.project.portal` will be returned.
(you can have a try on the market site of Portal).
and it also is useful for products like X-connector or Adobe Acrobat Sign to get the version before its original name is changed.

It's a long story and we would love to make this property name clearer if you have any suggestions

Reply:
Thanks a lot for your explanation. I now get the use case. 
I'd not rename the property today, as you will have to adapt the consuming backend also if we do it.

What I would do though, is improve the description. 
As I first looked at the property, I thought someone must manually raise this property whenever a new LTS 10 Portal is released (e.g. from 10.0.20 to 10.0.21). 
So I'd simply state that this is a 'max version' selector, and that all versions which are lower than the defined value, will be treated as legacy artifact. Therefore supplying 'legacy' group-/artifact-id for download.
Maybe even with a simple example. You can even supply the test as text-block if you like to present long content:
```
@JsonPropertyDescription("""
  This is a very long description
  Living in a java text-block
  Enriched with a nice example""")
```

Reply:
> is it inclusive or exclusive? or why do you use '10.1.0' to as value to describe the 10.0.X release train?

the initial design is inclusive while Wawa team confirmed with us that 10.0.19 is the last version that "com.axonivy.portal" is used for the group id of the portal app and portal guide.
But later, they wanted to keep the current group ID for version 10.0.xxx and not raise it to 10.1.xxx so they agreed to raise it to "10.1.0".

I will update the description, soon. Thanks for your feedback.